Update Button Styles + Brand Button

davinotdavid opened the pull request:
## Description of changes

- Added `BrandButton` component, which is the `BaseButton` with `type="brand"`
- Changed the `icon` slot to be `iconLeft` and/or `iconRight` for more flexibility
- Added a `variant` prop since we have `outline` / `filled` versions of buttons now
- Tweak the buttons according to the [Figma file](https://www.figma.com/design/1wGTkoIgRfyEbLeiW1XeIx/Desi%20gn-System--%3E-Components?node-id=3514-29487&m=dev)
- Removed the `SecondaryButton` component

> [!WARNING]
> This is a breaking change since we are removing the `SecondaryButton` and the default button if no options were passed was the secondary button.

## Known issues / Things to improve

- Link button is unchanged yet as it is TBD in the [Figma file](https://www.figma.com/design/1wGTkoIgRfyEbLeiW1XeIx/Desi%20gn-System--%3E-Components?node-id=3514-29487&m=dev)
- Design team's desire is to make a transparent background for all the outline buttons. I legitimate couldn't find a way until now to make a button that has: border radius, gradient on border _and_ is transparent. Not sure if this is a hard requirement but if it is, perhaps it can be done in a follow up as I spent way too much time already.

## Solves
https://github.com/thunderbird/services-ui/issues/113
https://github.com/thunderbird/services-ui/issues/114
https://github.com/thunderbird/services-ui/issues/100
https://github.com/thunderbird/services-ui/issues/101

MelissaAutumn reviewed (changes requested):
Looking good, just needs some minor changes:

* Could we add a smooth animation to the state changes? I find them a little harsh between states and worried that it might impact the design. 
* I found one tiny css tweak you can add to fix button heights
* Can you add a story for the outline button?

davinotdavid commented on src/elements/BaseButton.vue:
```diff
@@ -59,57 +81,187 @@ button:hover > .tooltip,
 
-.primary {
-  --colour-btn-border: var(--colour-service-primary-hover);
+.base {
```
For sure! Thanks!

MelissaAutumn commented:
Probably all button types, but we could ask @laurelterlesky as well. It's just a bit jarring for brand especially.

davinotdavid commented:
@MelissaAutumn cool! Added for in the styles for the `base-button` but had to change a little on how we deal with the outline variant borders. On the plus side, there's no margin trickery anymore and transitions look better. On the minus side, removed the gradient on the outline border of the primary button for that to work well (IMHO I couldn't see the gradient and this makes it much more maintainable).

davinotdavid commented:
@MelissaAutumn sure! Added `prefers-reduced-motion` for this component specifically but as a reminder, this will be tackled more globally in this ticket here https://github.com/thunderbird/services-ui/issues/138

And you are right! The border gradient was different from the Figma file, good catch. I colorpicked my way to get it closer as using the same values as in Figma doesn't produce the same results as the origin spots are not the same (also [this article was super cool](https://9elements.com/blog/gradient-angles-in-css-figma-and-sketch/)).

> Btw, for the outline buttons have you tried the background with a rgba, and background-color: transparent?

Yep, no luck there either. @radishmouse and I will spend some time this afternoon trying to accomplish this somehow but neither of us were successful so far.
